blog/2021-06-17-activity-timeouts.md

```diff
@@ -0,0 +1,258 @@
+---
+tags:
+  - Temporal
+  - architecture
+  - timeouts
+posted_on_: 2021-06-17T00:00:09Z
+slug: activity-timeouts
+title: 'The 4 Types of Activity Timeouts'
+author: swyx
+author_title: Head of Developer Experience
+author_image_url: https://avatars.githubusercontent.com/u/6764957?v=4
+release_version: V1.10.2
+---
+
+
+<!--truncate-->
+
+Part of the benefit of moving business logic to Temporal is how it implements retries and timeouts for activities in a standardized way. This has the effect of adding a reliability layer atop unreliable activities and workers, in a durable and scalable fashion. However, understanding the terminology can be a bit intimidating at first glance. 
+
+This post (together with the embedded talk) aims to give you a solid mental model on what each activity timeout does and when to use it.
+
+> Note: There are also [workflow timeouts](https://docs.temporal.io/docs/concepts/workflows#timeout-settings) and [retry policies](https://docs.temporal.io/docs/concepts/activities#retries) you can set. This post only deals with *activity* timeouts.
+
+## Talk Version: Whiteboard Session
+
+You can watch this 18 minute talk where our CEO Maxim Fateev explains the 4 Types of Activity Timeouts you see in Temporal.
+
+import { ResponsivePlayer } from '../src/components'
+
+<ResponsivePlayer url='https://www.youtube.com/watch?v=JK7WLK3ZSu8' />
+
+The rest of this post summarizes the main points covered in this whiteboard session, for those who prefer a written version.
+
+## Written Version: TL;DR
+
+![image](https://user-images.githubusercontent.com/6764957/122424123-0ab1e300-cfc1-11eb-82c5-cf8856a87242.png)
+
+Activities go through 3 main states in Temporal: 
+
+- Scheduled
+- Started
+- Closed
+
+There are 4 Timeouts in Temporal — 2 that are commonly used, and 2 that are only useful in specific cases:
+
+- `ScheduleToClose`: to limit maximum execution time including retries
+- `StartToClose`: to limit maximum execution time of a single invocation. **We recommend ALWAYS setting this!**
+- `Heartbeat`: *For long running activities*, to respond faster when a regular heartbeat fails to be recorded.
```

Comment:
```suggestion
- `Heartbeat`: *For long-running activities*, to respond faster when a regular heartbeat fails to be recorded.
```

@flossypurse: Hyphenating "long-running" doesn't match our current usage, but I'm recommending the change. Stet if you want to hold off for now.